pkg/clusterregistryconfig/flags.go

```diff
@@ -0,0 +1,350 @@
+	result.platformAllowlist = args.platformAllowlist
+
+	if !IsClusterRegistryConfigSetViaCLI(cmd) {
+		if !interactive.Enabled() {
+			return nil, nil
+		}
+	}
+
+	enableRegistriesConfig := IsClusterRegistryConfigSetViaCLI(cmd)
+
+	if !enableRegistriesConfig && interactive.Enabled() && isHostedCP {
+		updateRegistriesConfigValue, err := interactive.GetBool(interactive.Input{
+			Question: "Enable registries config",
+			Default:  enableRegistriesConfig,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value: %s", err)
+		}
+		enableRegistriesConfig = updateRegistriesConfigValue
+	}
+
+	if enableRegistriesConfig && interactive.Enabled() {
+		allowedRegistriesInputs, err := interactive.GetString(interactive.Input{
+			Question: "Allowed Registries",
+			Help:     cmd.Lookup("allowed-registries").Usage,
+			Default:  args.allowedRegistries,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for allowed registries: %s", err)
+		}
+		result.allowedRegistries = helper.HandleEmptyStringOnSlice(strings.Split(allowedRegistriesInputs, ","))
+
+		blockedRegistriesInputs, err := interactive.GetString(interactive.Input{
+			Question: "Blocked Registries",
+			Help:     cmd.Lookup("blocked-registries").Usage,
+			Default:  args.blockedRegistries,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for blocked registries: %s", err)
+		}
+		result.blockedRegistries = helper.HandleEmptyStringOnSlice(strings.Split(blockedRegistriesInputs, ","))
+
+		insecureRegistriesInputs, err := interactive.GetString(interactive.Input{
+			Question: "Insecure Registries",
+			Help:     cmd.Lookup("insecure-registries").Usage,
+			Default:  args.insecureRegistries,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for insecure registries: %s", err)
+		}
+		result.insecureRegistries = helper.HandleEmptyStringOnSlice(strings.Split(insecureRegistriesInputs, ","))
+
+		result.allowedRegistriesForImport, err = interactive.GetString(interactive.Input{
+			Question: "Allowed Registries For Import",
+			Help:     cmd.Lookup("allowed-registries-for-import").Usage,
+			Default:  args.allowedRegistriesForImport,
+			Validators: []interactive.Validator{
+				ocm.ValidateAllowedRegistriesForImport,
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for allowed registries for import: %s", err)
+		}
+
+		result.additionalTrustedCa, err = interactive.GetString(interactive.Input{
+			Question: "Registry Additional Trusted CA",
+			Help:     cmd.Lookup("registry-additional-trusted-ca").Usage,
+			Default:  args.additionalTrustedCa,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid certificate: %s", err)
+		}
+	}
+	if err = ocm.ValidateAllowedRegistriesForImport(result.allowedRegistriesForImport); err != nil {
+		return nil, fmt.Errorf("Expected valid allowed registries for import values: %v", err)
+	}
+
+	return result, err
```

Comment:
```suggestion
	return result, nil
```

Reply:
Thanks @ziccardi , updated with the new change :)